java_sdk_example/src/main/java/aws/sample/paymentcryptography/hmac/HMACService.java

```diff
@@ -1,86 +1,103 @@
 import java.util.logging.Logger;
 
 import org.apache.commons.codec.binary.Hex;
 import org.springframework.stereotype.Component;
 
-import com.amazonaws.services.paymentcryptography.model.Alias;
-import com.amazonaws.services.paymentcryptography.model.CreateKeyRequest;
-import com.amazonaws.services.paymentcryptography.model.Key;
-import com.amazonaws.services.paymentcryptography.model.KeyAlgorithm;
-import com.amazonaws.services.paymentcryptography.model.KeyAttributes;
-import com.amazonaws.services.paymentcryptography.model.KeyClass;
-import com.amazonaws.services.paymentcryptography.model.KeyModesOfUse;
-import com.amazonaws.services.paymentcryptography.model.KeyUsage;
-import com.amazonaws.services.paymentcryptographydata.model.GenerateMacRequest;
-import com.amazonaws.services.paymentcryptographydata.model.GenerateMacResult;
-import com.amazonaws.services.paymentcryptographydata.model.MacAlgorithm;
-import com.amazonaws.services.paymentcryptographydata.model.MacAttributes;
-import com.amazonaws.services.paymentcryptographydata.model.VerifyMacRequest;
-import com.amazonaws.services.paymentcryptographydata.model.VerifyMacResult;
-import com.amazonaws.util.StringUtils;
-
 import aws.sample.paymentcryptography.ControlPlaneUtils;
 import aws.sample.paymentcryptography.DataPlaneUtils;
 import aws.sample.paymentcryptography.ServiceConstants;
+import software.amazon.awssdk.services.paymentcryptography.model.Alias;
+import software.amazon.awssdk.services.paymentcryptography.model.CreateKeyRequest;
+import software.amazon.awssdk.services.paymentcryptography.model.Key;
+import software.amazon.awssdk.services.paymentcryptography.model.KeyAlgorithm;
+import software.amazon.awssdk.services.paymentcryptography.model.KeyAttributes;
+import software.amazon.awssdk.services.paymentcryptography.model.KeyClass;
+import software.amazon.awssdk.services.paymentcryptography.model.KeyModesOfUse;
+import software.amazon.awssdk.services.paymentcryptography.model.KeyUsage;
+import software.amazon.awssdk.services.paymentcryptographydata.model.GenerateMacRequest;
+import software.amazon.awssdk.services.paymentcryptographydata.model.GenerateMacResponse;
+import software.amazon.awssdk.services.paymentcryptographydata.model.MacAlgorithm;
+import software.amazon.awssdk.services.paymentcryptographydata.model.MacAttributes;
+import software.amazon.awssdk.services.paymentcryptographydata.model.VerifyMacRequest;
+import software.amazon.awssdk.services.paymentcryptographydata.model.VerifyMacResponse;
+import software.amazon.awssdk.utils.StringUtils;
 
 @Component
 public class HMACService {
 
-    public String getHMACKey() {
+    public String getHMACKey() throws InterruptedException, ExecutionException {
         Alias hmacKeyAlias = ControlPlaneUtils.getOrCreateAlias(ServiceConstants.HMAC_KEY_ALIAS);
 
-        if (!StringUtils.isNullOrEmpty(hmacKeyAlias.getKeyArn())) {
-            return hmacKeyAlias.getKeyArn();
+        if (!StringUtils.isBlank(hmacKeyAlias.keyArn())) {
```

Comment:
I noticed you call this an hmac key but it's ISO9797-1 method 3 key which is based on TDES not HMAC.

Reply:
done